### ISSUE TITLE
attention

### ISSUE THREAD

**Issue by dilekozkan** [yykzjh/PMFSNet#8]
The decoder feature is always upsampled (aligned in spatial dimensions) before being passed to the attention gate.
"it is correct but i couldnt see on the diagram of attention unet's article . (it seems like there is no up conv on previous layer's signal at decoder just before attention)
so the code is like: 
d4 = self.Up4(d5)
        s3 = self.Att4(gate=d4, skip_connection=e3)
        d4 = torch.cat((s3, d4), dim=1)
        d4 = self.UpConv4(d4)
        ...
        
is it true
![article copy](https://github.com/user-attachments/assets/98723d06-491e-4693-bf74-0665a95a87bd)
?
